src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,141 @@
+        }
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (index < size && index >= 0) {
+            internalArray[index] = value;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Wrong array index!");
+        }
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        T oldelement = null;
+        if (index < this.size && index >= 0) {
+            int subsiquentIndex = 0;
+            oldelement = get(index);
+            Object[] newFormedArray = new Object[internalArray.length];
+            for (int i = 0; i < size; i++) {
+                if (i != index) {
+                    newFormedArray[subsiquentIndex] = internalArray[i];
+                    subsiquentIndex++;
+                }
+            }
+            internalArray = newFormedArray;
+            size--;
+        } else {
+            throw new ArrayListIndexOutOfBoundsException("Wrong index");
+        }
+        return oldelement;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        T oldElement = null;
+        for (int i = 0; i < size; i++) {
+            if (internalArray[i] != null && internalArray[i].equals(element)) {
+                oldElement = element;
+                remove(i);
+                break;
+            } else if (internalArray[i] == null && internalArray[i] == element) {
+                remove(i);
+                break;
+            }
+        }
+        if (oldElement == null && element != null) {
+            throw new NoSuchElementException("This element hasn't found");
+        }
+        return oldElement;
     }
 
     @Override
     public int size() {
-        return 0;
+        return this.size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void grow() {
+        Object[] bufferArray = new Object[size + (size >> 1)];
+        System.arraycopy(internalArray, 0, bufferArray, 0, size);
+        internalArray = bufferArray;
+    }
+
+    private void growByCell(int index) {
+        Object[] arrayBeforeIndex = new Object[index];
+        Object[] arrayAfterIndex = new Object[size - index];
+        System.arraycopy(internalArray, 0, arrayBeforeIndex, 0, index);
+        System.arraycopy(internalArray, index, arrayAfterIndex, 0, size - index);
+        if (index < size && (size + 1) <= internalArray.length) {
+            System.arraycopy(arrayBeforeIndex, 0, internalArray, 0, arrayBeforeIndex.length);
+            System.arraycopy(arrayAfterIndex, 0, internalArray, index + 1, arrayAfterIndex.length);
+        } else if (index < size && (size + 1) > internalArray.length) {
+            grow();
+            System.arraycopy(arrayBeforeIndex, 0, internalArray, 0, arrayBeforeIndex.length);
+            System.arraycopy(arrayAfterIndex, 0, internalArray, index + 1, arrayAfterIndex.length);
+        }
```

Comment:
Don't create redundant variables 'arrayBeforeIndex' and 'arrayAfterIndex'. They are unnecessary and make the code less clean. Use System.arraycopy() directly to shift the elements after the index where the new element is to be inserted.